### PR TITLE
PS-1584 Fix config validation for ABS workspace

### PR DESCRIPTION
**Pull request by pivnicek**
https://keboola.atlassian.net/browse/PS-1584

### CODE REVIEW

**Comment by odinuv** [comment]
can we have a functional test for the entire abs workspace?

**Comment by pivnicek** [comment]
I think so. I can try adding it to this? https://github.com/keboola/docker-bundle/blob/master/Tests/Runner/WorkspaceProviderTest.php. Not quite sure where to put it

**Comment by odinuv** [comment]
i mean somthing like this https://github.com/keboola/docker-bundle/blob/6b958330300bb33958c25ba7eab707eaa4125e49/Tests/Runner/RunnerTest.php#L2507

**Comment by pivnicek** [comment]
ok,  I'll give that a shot

**Comment by pivnicek** [comment]
It'll have to go into a separate file though because of the Synapse requirement, but that shouldn't be a huge obstacle.

**Comment by pivnicek** [comment]
Ah, i will put it here https://github.com/keboola/docker-bundle/blob/6b958330300bb33958c25ba7eab707eaa4125e49/Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php. cool.

**Comment by pivnicek** [comment]
@odinuv something like that?

File: Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php [hunk 1]
```diff
@@ -151,4 +163,102 @@ public function testWorkspaceMapping()
...
+                            'tables' => [
+                                [
+                                    'source' => 'local-table-out',
+                                    'destination' => 'out.c-synapse-runner-test.new-table',
+                                ],
+                            ],
+                        ],
+                    ],
+                    'parameters' => [
+                        'operation' => 'copy-abs',
```
Comment (odinuv) [inline Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php:246]: i think it has something like `list` to check if the input files are there

Comment (pivnicek) [reply Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php:246]: :-( it doesn't, but apparently it doesn't even have `copy-abs`, I was sure I added it. https://github.com/keboola/runner-workspace-test/blob/master/src/ConfigDefinition.php#L21

Comment (pivnicek) [reply Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php:246]: I will add the list operation there

**Comment by JakubMatejka** [comment]
Pánové prosím je třeba to zprovoznit co nejdřív, ať se může dát ta transformace k testování 🙏

File: Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php [hunk 2]
```diff
@@ -151,4 +163,95 @@ public function testWorkspaceMapping()
...
+        $options->setComponentId('keboola.runner-workspace-abs-test');
+        $options->setConfigurationId($configId);
+        self::assertCount(0, $components->listConfigurationWorkspaces($options));
+        $output = $this->getContainerHandler()->getRecords();
+        $blobFound = false;
+        foreach ($output as $blobName) {
+            if ($blobName === '/data/files/in/abs-workspace-file.csv') {
+                $blobFound = true;
+            } else {
+                echo "\nFound blob " . json_encode($blobName) . " which is not abs-workspace-file.csv\n";
```
Comment (pivnicek) [inline Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php:251]: @odinuv this output `"\/tmp\/docker\/run-5fcf5048655848.32865018\/data\/in\/files\/213339795_run_5fcf503725f0c9.18705067\/run_5fcf503725f0c9.18705067"`. So the blob is there, but not sure best how to assert it

Comment (pivnicek) [reply Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php:251]: it's not named properly

**Comment by odinuv** [comment]
@pivnicek please stop - i'll do something

**Comment by pivnicek** [comment]
ok, sorry, I'll rebase later to get rid of my garbage

File: Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php [hunk 3]
```diff
@@ -169,22 +169,15 @@ public function testAbsWorkspaceMapping()
...
         $temp = new Temp();
         $temp->initRunFolder();
-        $csv = new CsvFile($temp->getTmpFolder() . '/upload.csv');
-        $csv->writeRow(['id', 'text']);
-        $csv->writeRow(['test1', 'test1']);
-        $this->getClient()->createTableAsync('in.c-synapse-runner-test', 'mytable', $csv);
+        file_put_contents($temp->getTmpFolder() . '/my-lovely-file.wtf', 'some data');
         $this->getClient()->uploadFile(
-            $csv->getPath(),
+            $temp->getTmpFolder() . '/my-lovely-file.wtf',
```
Comment (pivnicek) [inline Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php:177]: ntvl 🤦 diky maestro

**Review by JakubMatejka — APPROVED** [review]
👍

**Pull request merged**